lib/Defs.lhs

```diff
@@ -122,19 +122,71 @@ genFunctionToSubset ws = do
 (!) _ [] = Set.empty
 (!) i xs = xs !! (i `mod` length xs)
 
+-- The proposition are picked in the range (1, maximumArbitraryMFormPropositions) for MForm,
+-- and from (1, maximumArbitraryMFormPropositions) for Form.
+-- We cannot use the QuickCheck size because it would introduce a bias in the generation of Proposition values,
+-- where small sized examples can only choose small valued Propositions.
+maximumArbitraryMFormPropositions :: Int
+maximumArbitraryMFormPropositions = 32
+
+maximumArbitraryFormPropositions :: Int
+maximumArbitraryFormPropositions = 32
+
+arbitraryFormScaling :: Int 
+arbitraryFormScaling = 10
+
+arbitraryPropScaling :: Int
+arbitraryPropScaling = 5
+
 instance Arbitrary KrM where
   arbitrary = sized (\n -> do
     k <- choose (0, n)
     let ws = Set.fromList [0..k]
     r <- Map.fromList . zip [0..k] <$> vectorOf (k+1) (subsetOf ws)
-    v <- Map.fromList . zip [0..k] <$> vectorOf (k+1) arbitrary
+    v <- Map.fromList . zip [0..k] <$> vectorOf (k+1) (Set.fromList <$> scale (`div` arbitraryPropScaling) (listOf (choose (0, maximumArbitraryFormPropositions))))
```

Comment:
We need to pick propositions from the range `(0, maximumArbitraryFormPropositions)`